File: 11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxArea = 0;

        while(left < right){
            if(height[left] > height[right]){
                int area = height[right] * (right - left);
                if(area > maxArea) maxArea = area;
                right--;
            }
            else{
                int area = height[left] * (right - left);
                if(area > maxArea) maxArea = area;
                left++;
            }
        }

        return maxArea;
    } 
}

File: 11. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int maxSize = 0;

        while (left < right) {
            int dist = right - left;
            if (height[left] < height[right]) {
                maxSize = max(maxSize, height[left] * dist);
                left++;
            } else { 
                maxSize = max(maxSize, height[right] * dist);
                right--;
            }
        }
        return maxSize;
    }
}; 
